File: 2000 98 & ME/QuickStart/aspplus/samples/webforms/ctrlauth/Composition/VB/Composition2.vb
Option Strict Off

Imports System
Imports System.Web
Imports System.Web.UI
Imports System.Web.UI.WebControls

Namespace CompositionSampleControls

    Public Class Composition2VB : Inherits Control : Implements INamingContainer

        Public Property Value As Integer
           Get
               Dim Ctrl As TextBox = Controls(1)
               Return Int32.Parse(Ctrl.Text)
           End Get
           Set
               Dim Ctrl As TextBox = Controls(1)
               Ctrl.Text = Value.ToString()
           End Set
        End Property

        Protected Overrides Sub CreateChildControls()

           ' Add Literal Control

           Me.Controls.Add(New LiteralControl("<h3>Value: "))

           ' Add Textbox

           Dim Box As New TextBox
           Box.Text = "0"
           Me.Controls.Add(Box)

           ' Add Literal Control

           Me.Controls.Add(New LiteralControl("</h3>"))

           ' Add "Add" Button

           Dim AddButton As New Button
           AddButton.Text = "Add"
           AddHandler AddButton.Click, AddressOf AddBtn_Click
           Me.Controls.Add(AddButton)

           ' Add Literal Control

           Me.Controls.Add(New LiteralControl(" | "))

           ' Add "Subtract" Button

           Dim SubtractButton As New Button
           SubtractButton.Text = "Subtract"
           AddHandler SubtractButton.Click, AddressOf SubtractBtn_Click
           Me.Controls.Add(SubtractButton)

        End Sub

        Private Sub AddBtn_Click(Sender As Object, E As EventArgs)
           Me.Value = Me.Value + 1
        End Sub

        Private Sub SubtractBtn_Click(Sender As Object, E As EventArgs)
           Me.Value = Me.Value - 1
        End Sub

    End Class

End Namespace

File: 2000 98 & ME/QuickStart/aspplus/samples/webforms/ctrlauth/Composition/Composition2.cs
using System;
using System.Web;
using System.Web.UI;
using System.Web.UI.WebControls;

namespace CompositionSampleControls {

    public class Composition2 : Control, INamingContainer {

        public int Value {
           get {
               return Int32.Parse(((TextBox)Controls[1]).Text);
           }
           set {
               ((TextBox)Controls[1]).Text = value.ToString();
           }
        }

        protected override void CreateChildControls() {

           // Add Literal Control

           this.Controls.Add(new LiteralControl("<h3>Value: "));

           // Add Textbox

           TextBox box = new TextBox();
           box.Text = "0";
           this.Controls.Add(box);

           // Add Literal Control

           this.Controls.Add(new LiteralControl("</h3>"));

           // Add "Add" Button

           Button addButton = new Button();
           addButton.Text = "Add";
           addButton.Click += new EventHandler(this.AddBtn_Click);
           this.Controls.Add(addButton);

           // Add Literal Control

           this.Controls.Add(new LiteralControl(" | "));

           // Add "Subtract" Button

           Button subtractButton = new Button();
           subtractButton.Text = "Subtract";
           subtractButton.Click += new EventHandler(this.SubtractBtn_Click);
           this.Controls.Add(subtractButton);

        }

        private void AddBtn_Click(Object sender, EventArgs e) {
           this.Value++;
        }

        private void SubtractBtn_Click(Object sender, EventArgs e) {
           this.Value--;
        }        
    }    
}